__tests__/winston-cloudwatch.test.ts

```diff
@@ -0,0 +1,157 @@
+                Promise.resolve({
+                    logStreams: [
+                        {
+                            logStreamName: "test-stream",
+                            uploadSequenceToken: "test-token",
+                        },
+                    ],
+                })
+            )
+            .mockImplementationOnce(() => Promise.resolve({}));
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+        jest.useRealTimers();
+    });
+
+    it("should initialize with correct options", () => {
+        const options = {
+            logGroupName: "test-group",
+            logStreamName: "test-stream",
+            logBufferSize: 3,
+            timeout: 2000,
+        };
+
+        const instance = new WinstonCloudwatch(options);
+
+        expect(instance).toHaveProperty("logGroupName", options.logGroupName);
+        expect(instance).toHaveProperty("logStreamName", options.logStreamName);
+        expect(instance).toHaveProperty("logBufferSize", options.logBufferSize);
+        expect(instance).toHaveProperty("timeout", options.timeout);
+    });
+
+    it("should initialize with default values when not provided", () => {
+        const instance = new WinstonCloudwatch({
+            logGroupName: "test-group",
+            logStreamName: "test-stream",
+        });
+
+        expect(instance).toHaveProperty("logBufferSize", 500);
+        expect(instance).toHaveProperty("timeout", 1000);
+    });
+
+    it("should call cloudwatch functions", async () => {
+        const mockLog = {
+            level: "info",
+            message: "test message",
+            timestamp: Date.now(),
+        };
+
+        await transport.upload([mockLog]);
+
+        expect(mockCloudWatchLogsClient.prototype.send).toHaveBeenCalledWith(
+            expect.any(DescribeLogStreamsCommand)
+        );
+
+        expect(mockCloudWatchLogsClient.prototype.send).toHaveBeenCalledWith(
+            expect.any(PutLogEventsCommand)
+        );
+
+        expect(mockCloudWatchLogsClient.prototype.send).toHaveBeenCalledTimes(
+            2
+        );
+    });
+
+    it("should buffer logs and upload when buffer size is reached", async () => {
+        const mockLog = {
+            level: "info",
+            message: "test message",
+            timestamp: Date.now(),
+        };
+
+        WinstonCloudwatch.prototype.upload = jest.fn();
+
+        // Log messages until buffer is full
+        for (let i = 0; i < 3; i++) {
+            await transport.log(mockLog, () => {});
+        }
+
+        expect(WinstonCloudwatch.prototype.upload).toHaveBeenCalledTimes(1);
+    });
+
+    it("should buffer logs and not upload when buffer size is not reached", async () => {
+        const mockLog = {
+            level: "info",
+            message: "test message",
+            timestamp: Date.now(),
+        };
+
+        WinstonCloudwatch.prototype.upload = jest.fn();
+
+        // Log messages
+        for (let i = 0; i < 2; i++) {
```

Comment:
I'm just wondering, why will upload be called 0 times if our buffer size is set to 2?

Reply:
Yeah, that's true. I think that is a mistake.

Reply:
Resolved in https://github.com/initdsg/winston-cloudwatch/pull/1/commits/e1e101ed733ad2e881daa316029c5876b995f79f